locallibrary/catalog/forms.py

```diff
@@ -0,0 +1,20 @@
+import datetime
+
+from django import forms
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+class RenewBookForm(forms.Form):
+    renewal_date = forms.DateField(help_text="Enter a date between now and 4 weeks (default 3).")
+
+    def clean_renewal_date(self):
+        data = self.cleaned_data['renewal_date']
```

Comment:
thằng `cleaned_data` ở đâu ra em? nó lưu trữ những thông tin gì?

Reply:
cleaned_data là một API form của django có tác dụng làm sạch dữ liệu người dùng nhập, loại bỏ các dữ liệu không an toàn và chuyển dữ liệu về dạng dữ liệu tiêu chuẩn của django